### ISSUE TITLE
Scala TCK implementation

### ISSUE THREAD

**Issue by jrudolph** [lightbend/akkaserverless-java-sdk#564]
Part of #372

 * generate stubs from framework TCK protobuf definitions
 * implement stubs
 * setup test run

**Comment by jrudolph** [comment]
For the Java implementation of the TCK, generated Handlers and Providers have been committed currently.

@patriknw suggested to try to get codegen to work in the tck submodule to avoid having to manually update those when code generation changes.

**Comment by pvlugter** [comment]
> @patriknw suggested to try to get codegen to work in the tck submodule to avoid having to manually update those when code generation changes.

Yes, was just thinking about this too. And it could make sense to have the Java TCK implementation as a separate maven project to use codegen too.

**Comment by jrudolph** [comment]
> Yes, was just thinking about this too. And it could make sense to have the Java TCK implementation as a separate maven project to use codegen too.

I started doing it for the Java SDK but just using sbt (similar to what we have in the `java-gen-compilation-tests`). That would still cover the generation but not the maven plugin which seems acceptable and might be easier to maintain.

**Comment by pvlugter** [comment]
> I started doing it for the Java SDK but just using sbt (similar to what we have in the `java-gen-compilation-tests`). That would still cover the generation but not the maven plugin which seems acceptable and might be easier to maintain.

Good idea, that's better. Maven plugin is already covered.
